src/merkle_sum_tree/utils/csv_parser.rs

```diff
@@ -26,12 +26,12 @@ pub fn parse_csv_to_entries<P: AsRef<Path>>(path: P) -> Result<Vec<Entry>, Box<d
         entries.push(entry);
     }
 
-    // modulus from bn256 curve impl => https://github.com/privacy-scaling-explorations/halo2curves/blob/main/src/bn256/fr.rs#L38
-    const MODULUS_STR: &str = "30644e72e131a029b85045b68181585d2833e84879b9709143e1f593f0000001";
+    // For preventing overflow, we set the maximum value limit to 2^251 - 1
```

Comment:
Why not 252? I think we can get there using 84*3.

Reply:
Currently, we're using a single range table where `1 << 8` is applied to the decomposed columns for range checking. This means the overflow chip checks if the maximum number is (1 << 251) - 1. The following table might be useful for understanding:

|   	| a_0                                                            	| a_1  	| a_2  	| ... 	| a_30 	|
|---	|----------------------------------------------------------------	|------	|------	|-----	|------	|
| x 	| 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff 	| 0xff 	| 0xff 	| ... 	| 0xff 	|

In an ideal scenario, we could set the maximum value to half the modulus to avoid overflow, irrespective of the decomposed value. However, to achieve this, we'd need to create tables for range checking for each decomposed column.

|   	| a_0                                                                	| a_1  	| a_2  	| ... 	| a_32 	|
|---	|--------------------------------------------------------------------	|------	|------	|-----	|------	|
| x 	| 0x183227397098d014dc2822db40c0ac2e9419f4243cdcb848a1f0fac9f8000000 	| 0x18 	| 0x32 	| ... 	| 0x00 	|

However, I have my doubts about this approach. Each column from `a_1` to `a_30` requires a different range table. a_1 uses 0..24(0x18), a_2 uses 0..50(0x32), and so on.

One alternative that seems plausible is to use `0x1832` as the range for the table. While we could go for a larger number, we would need to set the maximum value to 2^n to fill all 252 bits. Additionally, as mentioned earlier, it seems challenging to exceed `0x183227`, which is between `1 << 20` and `1 << 21`.

The following table demonstrates how the maximum value can be checked for overflow using the 0x1832 range table:

|   	| a_0                                                                	| a_1    	| a_2    	| ... 	| a_17   	|
|---	|--------------------------------------------------------------------	|--------	|--------	|-----	|--------	|
| x 	| 0x1832183218321832183218321832183218321832183218321832183218321832 	| 0x1832 	| 0x1832 	| ... 	| 0x1832 	|

What are your thoughts on turning this idea into a new task?

Reply:
The idea of setting this check 

```
    // throw error if balance is larger than the modulus
    if balance_acc >= BigInt::parse_bytes(MODULUS_STR.as_bytes(), 16).unwrap() {
        return Err("Balance is larger than the modulus".into());
    }
```

was to warn the CEX that if they are building a MST which balance sum is larger than the modulus, it will end up in an overflow during proof generation.

Now I see that it would be more appropriate to check that the balance should not be larger than `MAX_VALUE_STR` which is the actual range check that we perform inside the circuit for the balance. 

But the value `MAX_VALUE_STR` depends on the value assigned to `ACC_COLS` and `MAX_BITS` which are generic to the circuit and might vary. Therefore I think it makes sense to keep this check against `MODULUS_STR`.  

In the case that `MODULUS < balance_acc < MAX_VALUE` the cex would be able to build the merkle sum tree and incur in an error during proof generation, but I think this is fine.


> In an ideal scenario, we could set the maximum value to half the modulus to avoid overflow, irrespective of the decomposed value

I didn't understand this part

Reply:
> But the value `MAX_VALUE_STR` depends on the value assigned to `ACC_COLS` and `MAX_BITS` which are generic to the circuit and might vary. Therefore I think it makes sense to keep this check against `MODULUS_STR`.
> 
> In the case that `MODULUS < balance_acc < MAX_VALUE` the cex would be able to build the merkle sum tree and incur in an error during proof generation, but I think this is fine.

I agree with that, it doesn't seem like the right time to settle this. Let's revert to this.

> > In an ideal scenario, we could set the maximum value to half the modulus to avoid overflow, irrespective of the decomposed value
> 
> I didn't understand this part

If we allow balances to go beyond half the modulus, meaning the overflow chip range check exceeds half the modulus, we could run into a worst-case scenario like the one below (considering a modulus of 255 for simplicity):

Let's walk through the 'merkle_prove_layer' in the circuit with some numbers:
https://github.com/summa-dev/summa-solvency/blob/b0fa4163474ac8ee1988342e070221f5586fb7a9/src/circuits/merkle_sum_tree.rs#L58-L81

leaf_balance = 28
path_element_balance[0] = 100, next_sum = 128
path_element_balance[1] = 128, next_sum = 1

In such a situation, the final next_sum value becomes 1, which wouldn't prevent an overflow situation.

This is the rationale behind setting the ideal maximum number for the overflow chip at half the modulus

Reply:
Reverted at [141d71c](https://github.com/summa-dev/summa-solvency/pull/33/commits/141d71c376f760cfccee77d0a21a6260f77f1866)